gnosis/eth/oracles/oracles.py

```diff
@@ -544,6 +546,48 @@ def get_price(self, token_address: str) -> float:
             )
 
 
+class SuperfluidOracle(PriceOracle):
+    def __init__(self, ethereum_client: EthereumClient, price_oracle: PriceOracle):
+        """
+        :param ethereum_client:
+        :param price_oracle: Price oracle to get the price for the components of Superfluid Tokens
+        """
+        self.ethereum_client = ethereum_client
+        self.w3 = ethereum_client.w3
+        self.price_oracle = price_oracle
+
+    @classmethod
+    def is_available(
+        cls,
+        ethereum_client: EthereumClient,
+    ) -> bool:
+        """
+        :param ethereum_client:
+        :return: `True` if Oracle is available for the EthereumClient provided, `False` otherwise
+        """
+        return (
+            ethereum_client.get_network() == EthereumNetwork.MATIC
+            or EthereumNetwork.XDAI
+            or EthereumNetwork.ARBITRUM
+            or EthereumNetwork.OPTIMISTIC
+        )
+
+    def get_price(self, token_address: str) -> float:
+        if not (Web3.isChecksumAddress(token_address)):
+            token_address = Web3.toChecksumAddress(token_address.lower())
```

Comment:
No need for these 2 lines

Reply:
Was getting errors in the testing, but I'll remove if the `token_address` here has already been checked when Oracle in cuse. For reference:
```
web3.exceptions.InvalidAddress: ('Web3.py only accepts checksum addresses. The software that gave you this non-checksum address should be considered unsafe, please file it as a bug on their platform. Try using an ENS name instead. Or, if you must accept lower safety, use Web3.toChecksumAddress(lower_case_address).', '0x1dbc1809486460dcd189b8a15990bca3272ee04e')
```

Reply:
That's because you are using not checksummed addresses in the tests